cmd/antrea-agent/agent.go

```diff
@@ -484,6 +496,18 @@ func run(o *Options) error {
 
 	go antreaClientProvider.Run(ctx)
 
+	if networkConfig.TrafficEncryptionMode == config.TrafficEncryptionModeIPSec &&
+		networkConfig.IPsecConfig.AuthenticationMode == config.IPsecAuthenticationModeCert {
+		go ipsecCertController.Run(stopCh)
+		klog.InfoS("Waiting for IPsec certificate to be configured")
+		if err := wait.PollUntil(time.Second, func() (bool, error) {
```

Comment:
this is a little obscure that people may not notice the order requirement between ipSecCertController and nodeRouteController. Perhaps injecting the dependency to nodeRouteController is more readable and reliable.

```
type IPsecCertificateManager interface {
	HasSynced() bool
}

Reply:
changed per suggestion. thanks!